websites/N/Nexon Logistics/presence.ts

```diff
@@ -0,0 +1,180 @@
+const presence = new Presence({
+		clientId: "1046638491435552819",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	let presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/m740VV3.jpg",
+	};
+	const [buttons, timestamp, privacy] = await Promise.all([
+			presence.getSetting<boolean>("buttons"),
+			presence.getSetting<boolean>("timestamp"),
+			presence.getSetting<boolean>("privacy"),
+		]),
+		pages: Record<string, PresenceData> = {
+			"/about": { details: "Viewing About Page" },
+			"/gallery": { details: "Viewing Gallery Page" },
+			"/events": { details: "Browsing Events Page" },
+			"/newsletter": { details: "Browsing Newsletter" },
+			"/rules": { details: "Reading the rules" },
+			"/PrivacyPolicy": { details: "Reading Privacy Policy" },
+			"/Terms&Condition": { details: "Reading Terms&Condition" },
+			"/leaderboard": { details: "Viewing Leaderboard" },
+		};
+
+	for (const [path, data] of Object.entries(pages)) {
+		if (document.location.pathname.includes(path))
+			presenceData = { ...presenceData, ...data };
+	}
+
+	switch (document.location.host) {
+		case "nexonlogistics.com":
+			switch (true) {
+				case document.location.pathname.includes("/newsletter/"):
```

Comment:
Instead of using `.includes()` you could use a RegEx to match the pattern which is more secure than using `.includes()` all the time